homeassistant/components/dlna_dmr/config_flow.py

```diff
@@ -408,3 +446,59 @@ def _is_ignored_device(discovery_info: ssdp.SsdpServiceInfo) -> bool:
         return True
 
     return False
+
+
+def _is_dmr_device(discovery_info: ssdp.SsdpServiceInfo) -> bool:
+    """Determine if discovery is a complete DLNA DMR device.
+
+    Use the discovery_info instead of DmrDevice.is_profile_device to avoid
+    contacting the device again.
+    """
+    # Abort if the device doesn't support all services required for a DmrDevice.
+    discovery_service_list = discovery_info.upnp.get(ssdp.ATTR_UPNP_SERVICE_LIST)
+    if not discovery_service_list:
+        return False
+
+    services = discovery_service_list.get("service")
+    if not services:
+        discovery_service_ids: set[str] = set()
+    elif isinstance(services, list):
+        discovery_service_ids = {service.get("serviceId") for service in services}
+    else:
+        # Only one service defined (etree_to_dict failed to make a list)
+        discovery_service_ids = {services.get("serviceId")}
+
+    if not DmrDevice.SERVICE_IDS.issubset(discovery_service_ids):
+        return False
+
+    return True
+
+
+async def _async_get_mac_address(hass: HomeAssistant, host: str) -> str | None:
+    """Get mac address from host name, IPv4 address, or IPv6 address."""
+    # Help mypy, which has trouble with the async_add_executor_job + partial call
+    mac_address: str | None
+    # getmac has trouble using IPv6 addresses as the "hostname" parameter so
+    # assume host is an IP address, then handle the case it's not.
+    try:
+        ip_addr = ip_address(host)
+    except ValueError:
+        mac_address = await hass.async_add_executor_job(
+            partial(get_mac_address, hostname=host)
```

Comment:
Is this data not available from the upnp data?

Reply:
Unfortunately not in a standardised way. Some devices will report it in a manufacturer-specific field of the UPnP device description, but it's so uncommon and and the field name varies so it is effectively useless.